ports/esp32/machine_uart.c

```diff
@@ -275,9 +276,20 @@ static void mp_machine_uart_init_helper(machine_uart_obj_t *self, size_t n_args,
         check_esp_err(uart_get_word_length(self->uart_num, &uartcfg.data_bits));
         check_esp_err(uart_get_parity(self->uart_num, &uartcfg.parity));
         check_esp_err(uart_get_stop_bits(self->uart_num, &uartcfg.stop_bits));
+        bool recreate_event_task = false;
+        if (self->uart_event_task != NULL) {
+            vTaskDelete(self->uart_event_task);
+            self->uart_event_task = NULL;
+            recreate_event_task = true;
+        }
         check_esp_err(uart_driver_delete(self->uart_num));
+        self->uart_queue = NULL;
         check_esp_err(uart_param_config(self->uart_num, &uartcfg));
-        check_esp_err(uart_driver_install(self->uart_num, self->rxbuf, self->txbuf, 0, NULL, 0));
+        check_esp_err(uart_driver_install(self->uart_num, self->rxbuf, self->txbuf, UART_QUEUE_SIZE, &self->uart_queue, 0));
+        if (recreate_event_task) {
+            xTaskCreatePinnedToCore(uart_event_task, "uart_event_task", 2048, self,
+                ESP_TASKD_EVENT_PRIO, (TaskHandle_t *)&self->uart_event_task, MP_TASK_COREID);
```

Comment:
Is it worth putting this in a (inline) helper function, now that it's called twice in this file?